ginivision/src/main/java/net/gini/android/vision/internal/ui/AlertDialogFragment.java

```diff
@@ -0,0 +1,107 @@
+            @StringRes final int message, @StringRes final int positiveButtonTitle,
+            @StringRes final int negativeButtonTitle, @StringRes final int dialogId) {
+        final Bundle args = new Bundle();
+        args.putInt(ARG_TITLE, title);
+        args.putInt(ARG_MESSAGE, message);
+        args.putInt(ARG_POSITIVE_BUTTON_TITLE, positiveButtonTitle);
+        args.putInt(ARG_NEGATIVE_BUTTON_TITLE, negativeButtonTitle);
+        args.putInt(ARG_DIALOG_ID, dialogId);
+        final AlertDialogFragment fragment = new AlertDialogFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        readArguments();
+    }
+
+    private void readArguments() {
+        final Bundle arguments = getArguments();
+        if (arguments == null) {
+            return;
+        }
+        mTitle = arguments.getInt(ARG_TITLE);
+        mMessage = arguments.getInt(ARG_MESSAGE);
+        mPositiveButtonTitle = arguments.getInt(ARG_POSITIVE_BUTTON_TITLE);
+        mNegativeButtonTitle = arguments.getInt(ARG_NEGATIVE_BUTTON_TITLE);
+        mDialogId = arguments.getInt(ARG_DIALOG_ID);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(final Bundle savedInstanceState) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        if (mTitle > 0) {
+            builder.setTitle(mTitle);
+        }
+        if (mMessage > 0) {
+            builder.setMessage(mMessage);
+        }
+        if (mPositiveButtonTitle > 0) {
+            builder.setPositiveButton(mPositiveButtonTitle, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(final DialogInterface dialog, final int which) {
+                    final FragmentActivity activity = getActivity();
+                    if (activity instanceof AlertDialogFragmentListener) {
+                        ((AlertDialogFragmentListener) activity).onPositiveButtonClicked(
+                                dialog, mDialogId);
+                    }
+                }
+            });
+        }
+        if (mNegativeButtonTitle > 0) {
+            builder.setNegativeButton(mNegativeButtonTitle, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(final DialogInterface dialog, final int which) {
+                    final FragmentActivity activity = getActivity();
+                    if (activity instanceof AlertDialogFragmentListener) {
+                        ((AlertDialogFragmentListener) activity).onNegativeButtonClicked(dialog,
+                                mDialogId);
+                    }
+                }
+            });
+        }
+        final AlertDialog alertDialog = builder.create();
+        alertDialog.setCanceledOnTouchOutside(false);
```

Comment:
This might be optional in some cases, if the possibility to omit the negative and positive button is offered I kinda think it should be possible to cancel the dialog.